daemon/mgr/container_types.go

```diff
@@ -64,6 +64,11 @@ func (c *Container) ID() string {
 	return c.meta.ID
 }
 
+// Image returns container's image name.
+func (c *Container) Image() string {
+	return c.meta.Config.Image
```

Comment:
Currently, In mgr's code, there are so many pointer calling. As a result, if struct has many pointers, there will be more potential panic there. I tested some and find some obvious panic, such as https://github.com/alibaba/pouch/issues/352.

Maybe we need to add more rule of contractual programming right now to guarantee the quality of code.

Reply:
Many pointers is not a problem, We don't need to be afraid of this. such as "c.meta.Config.Image", We must guarantee the "C" (Container) is valid at the origin position only, so the pointers don't be nil.

Reply:
> We must guarantee the "C" (Container) is valid at the origin position only

👍 

This conversation is not blocking this pull request.